src/CsrValidation/java/lib/src/main/java/com/microsoft/intune/scepvalidation/IntuneClient.java

```diff
@@ -417,4 +437,38 @@ private CloseableHttpClient getCloseableHttpClient()
 
         return this.httpClientBuilder.build();
     }
+    
+    private void setProxy()
+    {
+        if(proxyHost != null && !proxyHost.isEmpty() &&
+           proxyPort != null && !proxyPort.isEmpty())
+         {
+             this.authClient.SetProxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyHost, Integer.parseInt(proxyPort))));
+
+             this.log.info("Setting IntuneClient ProxyHost:" + proxyHost + " ProxyPort:" + proxyPort);
+             this.httpClientBuilder = HttpClients.custom().setProxy(new HttpHost(proxyHost, Integer.parseInt(proxyPort)));
+             
+             if(proxyUser != null && !proxyUser.isEmpty() &&
+                proxyPass != null && !proxyPass.isEmpty())
+              {
+                 this.log.info("Setting IntuneClient Proxy to use Basic Authentication.");
+                 
+                 Credentials credentials = new UsernamePasswordCredentials(proxyUser, proxyPass);
+                 CredentialsProvider credsProvider = new BasicCredentialsProvider();
+                 credsProvider.setCredentials( new AuthScope(proxyHost,Integer.parseInt(proxyPort)), credentials);
```

Comment:
This not likely a costly operation but why do it twice?  I suggest just store it on line 449 and then use it here.